### ISSUE TITLE
Plugin fails in Android-Studio 0.4.4

### ISSUE THREAD

**Issue by saadfarooq** [a-iv/IntelliJ-Time-Tracker#2]
Plugin fails in AS-0.4.4 (which is based on the IntelliJ EAP codebase).

**Comment by a-iv** [comment]
@saadfarooq Can you provide traceback?

**Comment by saadfarooq** [comment]
Hope this helps...

```
0: 0
java.lang.ArrayIndexOutOfBoundsException: 0
    at com.intellij.ide.ui.laf.darcula.ui.DarculaSpinnerUI.replaceEditor(DarculaSpinnerUI.java:65)
    at javax.swing.plaf.basic.BasicSpinnerUI$Handler.propertyChange(BasicSpinnerUI.java:955)
    at java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:339)
    at java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:276)
    at java.awt.Component.firePropertyChange(Component.java:8287)
    at javax.swing.JSpinner.setEditor(JSpinner.java:494)
    at com.toedter.components.JSpinField.<init>(Unknown Source)
    at com.toedter.components.JSpinField.<init>(Unknown Source)
    at com.toedter.calendar.JYearChooser.<init>(Unknown Source)
    at com.toedter.calendar.JCalendar.<init>(Unknown Source)
    at com.toedter.calendar.JCalendar.<init>(Unknown Source)
    at com.toedter.calendar.JDateChooser.<init>(Unknown Source)
    at com.toedter.calendar.JDateChooser.<init>(Unknown Source)
    at com.toedter.calendar.JDateChooser.<init>(Unknown Source)
    at com.toedter.calendar.JDateChooser.<init>(Unknown Source)
    at net.roarsoftware.tracker.ui.HistoryWindow.<init>(HistoryWindow.java:133)
    at net.roarsoftware.tracker.TimeTrackerPlugin.projectOpened(TimeTrackerPlugin.java:90)
    at net.roarsoftware.tracker.TimeTrackerPlugin$1.projectOpened(TimeTrackerPlugin.java:53)
    at com.intellij.openapi.project.impl.ProjectManagerImpl.fireProjectOpened(ProjectManagerImpl.java:1068)
    at com.intellij.openapi.project.impl.ProjectManagerImpl.openProject(ProjectManagerImpl.java:422)
    at com.intellij.openapi.project.impl.ProjectManagerImpl.loadAndOpenProject(ProjectManagerImpl.java:513)
    at com.intellij.ide.impl.ProjectUtil.openProject(ProjectUtil.java:181)
    at com.intellij.ide.RecentProjectsManager.doOpenProject(RecentProjectsManager.java:45)
    at com.intellij.ide.RecentProjectsManagerBase.doReopenLastProject(RecentProjectsManagerBase.java:345)
    at com.intellij.ide.RecentProjectsManagerBase$MyAppLifecycleListener.appStarting(RecentProjectsManagerBase.java:370)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:114)
    at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:228)
    at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:219)
    at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:209)
    at com.intellij.util.messages.impl.MessageBusImpl.access$000(MessageBusImpl.java:43)
    at com.intellij.util.messages.impl.MessageBusImpl$1.invoke(MessageBusImpl.java:131)
    at com.sun.proxy.$Proxy39.appStarting(Unknown Source)
    at com.intellij.idea.IdeaApplication.loadProject(IdeaApplication.java:284)
    at com.intellij.idea.IdeaApplication.access$500(IdeaApplication.java:56)
    at com.intellij.idea.IdeaApplication$IdeStarter$2.run(IdeaApplication.java:261)
    at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:319)
    at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:209)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:715)
    at java.awt.EventQueue.access$400(EventQueue.java:82)
    at java.awt.EventQueue$2.run(EventQueue.java:676)
    at java.awt.EventQueue$2.run(EventQueue.java:674)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:86)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:685)
    at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:697)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:524)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:335)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:296)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:211)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:196)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:188)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)
```

**Comment by a-iv** [comment]
This is an issue in Darcula theme.
Note that IntelliJ theme also uses Darcula.

I'm not an author of this plugin. And for now I can't fix this issue.
You can try to change color theme.

**Comment by saadfarooq** [comment]
This is the trace if I change to the default theme.

```
com/intellij/peer/PeerFactory: com/intellij/peer/PeerFactory
java.lang.NoClassDefFoundError: com/intellij/peer/PeerFactory
    at net.roarsoftware.tracker.TimeTrackerPlugin.projectOpened(TimeTrackerPlugin.java:93)
    at net.roarsoftware.tracker.TimeTrackerPlugin$1.projectOpened(TimeTrackerPlugin.java:53)
    at com.intellij.openapi.project.impl.ProjectManagerImpl.fireProjectOpened(ProjectManagerImpl.java:1068)
    at com.intellij.openapi.project.impl.ProjectManagerImpl.openProject(ProjectManagerImpl.java:422)
    at com.intellij.openapi.project.impl.ProjectManagerImpl.loadAndOpenProject(ProjectManagerImpl.java:513)
    at com.intellij.ide.impl.ProjectUtil.openProject(ProjectUtil.java:181)
    at com.intellij.ide.RecentProjectsManager.doOpenProject(RecentProjectsManager.java:45)
    at com.intellij.ide.RecentProjectsManagerBase.doReopenLastProject(RecentProjectsManagerBase.java:345)
    at com.intellij.ide.RecentProjectsManagerBase$MyAppLifecycleListener.appStarting(RecentProjectsManagerBase.java:370)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:114)
    at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:228)
    at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:219)
    at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:209)
    at com.intellij.util.messages.impl.MessageBusImpl.access$000(MessageBusImpl.java:43)
    at com.intellij.util.messages.impl.MessageBusImpl$1.invoke(MessageBusImpl.java:131)
    at com.sun.proxy.$Proxy39.appStarting(Unknown Source)
    at com.intellij.idea.IdeaApplication.loadProject(IdeaApplication.java:284)
    at com.intellij.idea.IdeaApplication.access$500(IdeaApplication.java:56)
    at com.intellij.idea.IdeaApplication$IdeStarter$2.run(IdeaApplication.java:261)
    at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:319)
    at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:209)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:715)
    at java.awt.EventQueue.access$400(EventQueue.java:82)
    at java.awt.EventQueue$2.run(EventQueue.java:676)
    at java.awt.EventQueue$2.run(EventQueue.java:674)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:86)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:685)
    at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:697)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:524)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:335)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:296)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:211)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:196)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:188)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)
Caused by: java.lang.ClassNotFoundException: com.intellij.peer.PeerFactory PluginClassLoader[Time Tracker, 0.3]
    at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:68)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:247)
    ... 41 more
```

**Comment by a-iv** [comment]
This issue has been fixed in 0caf9549961ab8a7ebed5c2806c6f28a0100e257.

The current master (cf302a21920a5be9c55e85899ea4a513c1df257b) was successfully tested on the latest stable version of Android Studio (2020.3.1).

**Comment by saadfarooq** [comment]
Great, thanks man!!
